File: 프로그래머스/lv0/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):
    answer = [0,0]
    for k in keyinput:
        temp1 = answer[1]
        temp0 = answer[0]
        if k == "up": 
            answer[1] += 1
        elif k == "down":
            answer[1] -= 1
        elif k == "left":
            answer[0] -= 1
        else:
            answer[0] += 1
        if board[1]//2*(-1)>answer[1] or answer[1]>board[1]//2:
            answer[1] = temp1
        elif board[0]//2*(-1)>answer[0] or answer[0]>board[0]//2:
            answer[0] = temp0

    return answer

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.java
import java.util.*;
class Solution {
    public int[] solution(String[] keyinput, int[] board) {
        HashMap<String, int[]> direction = new HashMap<>();
        direction.put("up", new int[]{0, 1});
        direction.put("down", new int[]{0, -1});
        direction.put("left", new int[]{-1, 0});
        direction.put("right", new int[]{1, 0});
        
        int w = (int) board[0] / 2;
        int h = (int) board[1] / 2;
        
        int x = 0, y = 0;
        for(String key : keyinput) {
            int dx = direction.get(key)[0];
            int dy = direction.get(key)[1];
            int nx = x + dx;
            int ny = y + dy;
            
            if(nx<-w || nx > w || ny<-h || ny>h) continue;
            
            x = nx;
            y = ny;
        }
        return new int[]{x, y};
    }
}